cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -189,10 +189,8 @@ protected void doShutdown() throws Exception {
   protected void runOneIteration() throws Exception {
     LOG.trace("Performing heartbeat publishing in Stream service instance {}", instanceId);
     ImmutableMap.Builder<Id.Stream, Long> sizes = ImmutableMap.builder();
-    Id.Namespace namespace = Id.Namespace.from(Constants.DEFAULT_NAMESPACE);
-    //TODO: use listStreams() across all namespaces, not just default namespace.
-    for (StreamSpecification streamSpec : streamMetaStore.listStreams(namespace)) {
-      Id.Stream streamId = Id.Stream.from(namespace, streamSpec.getName());
+    for (Map.Entry<Id.Namespace, StreamSpecification> streamSpecEntry : streamMetaStore.listStreams().entries()) {
```

Comment:
I've discussed with @jguery that listing all streams from mds for heartbeat is not a very good idea. A given heartbeat should only has stream size that is collected (meaning events for a given stream have been sent to this stream handler instance). They are more or less the same right now (not exactly though, because if there nothing written to a stream, ideally there is no heartbeat needed). However, when we add resource allocation for stream handler such that each handler may be only handling a subset of streams (which we will), then this logic is not good.